src/devices/machine/ibmps2.h

```diff
@@ -0,0 +1,265 @@
+// license:BSD-3-Clause
+// copyright-holders:Wilbert Pol, Miodrag Milanovic, Carl
+#ifndef MAME_MACHINE_PS2_H
+#define MAME_MACHINE_PS2_H
+
+#include "machine/am9517a.h"
+#include "machine/at_keybc.h"
+#include "machine/ins8250.h"
+#include "machine/mc146818.h"
+#include "machine/nvram.h"
+#include "machine/pic8259.h"
+#include "machine/pit8253.h"
+#include "machine/ram.h"
+#include "machine/upd765.h"
+#include "machine/pc_lpt.h"
+#include "machine/ibm72x7377.h"
+#include "bus/mca/mca.h"
+#include "bus/pc_kbd/keyboards.h"
+#include "bus/pc_kbd/pc_kbdc.h"
+#include "sound/spkrdev.h"
+#include "softlist.h"
+
+class ps2_mb_device : public device_t
+{
+public:
+	ps2_mb_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock,
+		uint16_t sram_size, uint16_t planar_id)
+		: ps2_mb_device(mconfig, tag, owner, clock)
+	{
+		m_sram_size = sram_size;
+		m_planar_id = planar_id;
+	}
+	ps2_mb_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock);
```

Comment:
Is this an abstract class, or a concrete class?  If it’s abstract, the constructors should all be protected.  Also, any device class used as a base for other device classes needs to have protected constructors that accept the device type.